extensions/amp-story/1.0/amp-story-page-attachment.js

```diff
@@ -0,0 +1,383 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Action, getStoreService} from './amp-story-store-service';
+import {Layout} from '../../../src/layout';
+import {closest} from '../../../src/dom';
+import {dev} from '../../../src/log';
+import {htmlFor} from '../../../src/static-template';
+import {resetStyles, setImportantStyles, toggle} from '../../../src/style';
+
+
+/** @const {number} */
+const TOGGLE_THRESHOLD_PX = 50;
+
+/**
+ * @enum {number}
+ */
+const AttachmentState = {
+  CLOSED: 0,
+  DRAGGING_TO_CLOSE: 1,
+  DRAGGING_TO_OPEN: 2,
+  OPEN: 3,
+};
+
+/**
+ * Drawer's template.
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const getTemplateEl = element => {
+  return htmlFor(element)`
+    <div class="i-amphtml-story-page-attachment">
+      <div class="i-amphtml-story-page-attachment-header">
+        <span
+            class="i-amphtml-story-page-attachment-close-button" role="button">
+        </span>
+      </div>
+      <div class="i-amphtml-story-page-attachment-container">
+        <div class="i-amphtml-story-page-attachment-content"></div>
+      </div>
+    </div>`;
+};
+
+/**
+ * AMP Story page attachment.
+ */
+export class AmpStoryPageAttachment extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.containerEl_ = null;
+
+    /** @private {?Element} */
+    this.contentEl_ = null;
+
+    /** @private {!AttachmentState} */
+    this.state_ = AttachmentState.CLOSED;
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = getStoreService(this.win);
+
+    /** @private {boolean} */
+    this.ignoreCurrentSwipeYGesture_ = false;
+
+    /** @private {!Object} */
+    this.touchEventState_ = {
+      startX: 0,
+      startY: 0,
+      lastY: 0,
+      swipingUp: null,
+      isSwipeY: null,
+    };
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout === Layout.NODISPLAY;
+  }
+
+  /** @override */
+  prerenderAllowed() {
+    return false;
+  }
+
+  /** @override */
+  buildCallback() {
+    // TODO: maybe render the header in Shadow DOM?
+    const templateEl = getTemplateEl(this.element);
+
+    this.containerEl_ = dev().assertElement(
+        templateEl.querySelector('.i-amphtml-story-page-attachment-container'));
+    this.contentEl_ = dev().assertElement(this.containerEl_
+        .querySelector('.i-amphtml-story-page-attachment-content'));
+
+    while (this.element.firstChild) {
```

Comment:
s/`while`/`if` ?

Reply:
It's actually copying all the direct children of `this.element` (what's provided by the publisher) one by one into our own drawer like UI without cloning them, which the `copyChildren` helper would do.